nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -661,6 +665,15 @@ sub generateXserverConfig {
     mkpath($outDir, 0, 0755);
     write_file($fn, $hwConfig);
 
+    $fn = "$outDir/flake.nix";
+    if ($flake && ($force || ! -e $fn)) {
+        print STDERR "writing $fn...\n";
+        mkpath($outDir, 0, 0755);
+        write_file($fn, <<EOF);
+        @flake@
+EOF
+    }
```

Comment:
```suggestion
    if ($flake) {
        if ($force || ! -e $fn) {
            print STDERR "writing $fn...\n";
            mkpath($outDir, 0, 0755);
            write_file($fn, <<EOF);
            @flake@
EOF
        } else {
            print STDERR "warning: not overwriting existing $fn\n";
        }
    }
```
missed adding the warning earlier